spec/openhab/core/items/item_spec.rb

```diff
@@ -15,6 +15,31 @@
     end
   end
 
+  describe "#all_groups" do
+    it "includes all ancestor groups" do
+      items.build do
+        group_item "MainFloor", groups: [House] do
+          switch_item LightSwitch2
+        end
+      end
+
+      expect(LightSwitch2.all_groups).to eql [MainFloor, House]
```

Comment:
Should this use `match_array` instead of `eql`?

Reply:
eql is more explicit, requiring the original order. match_array doesn't care about order. I wanted to be sure that the immediate parent groups are first, followed by each generation